freight/providers/pipeline.py

```diff
@@ -408,6 +409,13 @@ def run_step_stateful_set(
                 ss.metadata.annotations[k] = v
                 ss.spec.template.metadata.annotations[k] = v
 
+            # Rolling update will update only pods with ordinal >= than the `partition` attribute
+            # https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#rolling-out-a-canary
+            if num_of_canary_instances:
+                ss.spec.update_strategy.rolling_update.partition = ss.status.replicas - num_of_canary_instances
+            else:
+                ss.spec.update_strategy.rolling_update.partition = 0
+
```

Comment:
I think we'd wanna set this back to `0` at the end of the deploy. Otherwise applying any changes from `kubectl` or `sentry-kube` are going to apply only to the 1 instance or whatever was used for a canary. Since this update_strategy is going to persist.

Reply:
Absolutely right, we are aware of it. But if I understand the docs correctly (https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#phased-roll-outs), setting it back to 0 at the end of the deploy would roll the update to all instances.